md2html.sh

```diff
@@ -16,12 +18,36 @@ while getopts "hl" opt; do
 done
 
 for file in "$@"; do
-	pandoc -f markdown_github "${file}" -t asciidoc -o "${file%.md}.tmp.adoc"
-	if [ "$convert_link" -ne 0 ]; then
-		sed -i -e "s/\.md\>/\.html/g" "${file%.md}.tmp.adoc"
-	fi
-	touch -r "${file}" "${file%.md}.tmp.adoc"
-	TZ=UTC asciidoc -o "${file%.md}.html" -a footer-style=none -a toc2 -a source-highlighter=highlight "${file%.md}.tmp.adoc"
-	tr -d '\r' < "${file%.md}.html" > "${file%.md}.tmp.adoc"
-	mv "${file%.md}.tmp.adoc" "${file%.md}.html"
+        md2html_tool=""
+        # RHEL 9 in brew cannot use pandoc, hence trying kramown first
+        if rpm -q rubygem-kramdown-parser-gfm > /dev/null; then
+          md2html_tool=kramdown
+        elif rpm -q pandoc >/dev/null; then
```

Comment:
I think it is better to check for the availability of the actual commands, rather than for the presence of the RPMs that contain them, because you ultimately care about the commands, not the RPMs. Repeating an example: https://github.com/rear/rear/blob/6a3d0b4d5e73c69a62ce0bd209b2b38ffb462569/doc/Makefile#L10

Reply:
I remember about checking with `type`, however, the issue here is that for the kramdown, the script needs the kramdown-parser-gfm RPM installed, not just kramdown. I can check it by verifying the files that the `rubygem-kramdown-parser-gfm` package or the `kramdown-parser-gfm` gem installs, however checking files existance does not seem as reliable as checking for the RPM. Does that sound correct?
I then used the same check for the rpm for pandoc too intuitively for consistency, however, I can use `type` for pandoc as before, will that help?

Reply:
Hmm, can you perform a dummy invocation of `kramdown` that will verify if the gfm parser is installed?

Reply:
apparently this won't work, because if the gfm parser is not installed, `kramdown -x parser-gfm` will just print the message `Couldn't load extension parser-gfm, ignoring` and continue as if nothing wrong has happened. `gem info kramdown-parser-gfm` also does not return any error if the gem is not available. So keep the code as it is now because I don't see any better option.

Reply:
Yes, e.g. run kramdown -i GFM README.md, this either passes with 0 rc, or fails with Error: kramdown has no parser to handle the specified input format: GFM and 1 rc.
I will use this instead of the rpm -ql check. Thank you

Reply:
good, I missed the difference between `-i` and `-x`. You can use `kramdown -i GFM < /dev/null`, no need to use a real file.